eth-repository/src/main/kotlin/org/apache/tuweni/eth/repository/BlockchainIndex.kt

```diff
@@ -483,10 +483,8 @@ class BlockchainIndex(private val indexWriter: IndexWriter) : BlockchainIndexWri
       val topDocs = searcher!!.search(query, HITS)
 
       val docs = mutableListOf<Document>()
-      for (hit in topDocs.scoreDocs) {
-        val doc = searcher.doc(hit.doc, setOf("_id") + fields)
-        docs += doc
-      }
+      val doc = searcher.doc(topDocs.scoreDocs.elementAt(0).doc, setOf("_id") + fields)
+      docs += doc
```

Comment:
Maybe just return the doc instead of a list here.

Reply:
It will change the return type from List<Document> to a single Document, is that okay?

Reply:
Looks ok from here.

Reply:
When I make the above change, I get this error:
Task :eth-repository:compileKotlin FAILED
e: /home/gradle/eth-repository/src/main/kotlin/org/apache/tuweni/eth/repository/BlockchainIndex.kt: (485, 14): Type mismatch: inferred type is Document! but List<Document> was expected

FAILURE: Build failed with an exception.